.github/workflows/tests.yml

```diff
@@ -45,3 +45,54 @@ jobs:
       - name: Run unit tests
         run: ansible-test units --venv
         working-directory: ansible_collections/ansible/eda
+
+  integration-tests:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        python-version:
+          - "3.11"
+
+    env: 
+      EDA_CONTROLLER_HOST: "https://localhost:8443"
+      EDA_CONTROLLER_USERNAME: "admin"
+      EDA_CONTROLLER_PASSWORD: "testpass"
+      EDA_CONTROLLER_VERIFY_SSL: "False"
+
+    steps:
+      - name: Checkout ansible/eda
+        uses: actions/checkout@v3
+        with:
+          repository: ansible/event-driven-ansible
+          path: ansible_collections/ansible/eda
+          ref: api-module
+
+      - name: Checkout ansible/eda-server
+        uses: actions/checkout@v3
+        with:
+          repository: ansible/eda-server
+          path: eda-server # Note: Specify a separate path for this repository
+       
+      - name: Run API in background
+        run: |
+          docker compose -p eda -f docker-compose-stage.yaml pull
```

Comment:
This docker-compose used by default the latest build from main. This works for now, but we will need to release tagged images of eda-server and bind that tag to the version of the collection, otherwise the tests would fail in the future with breaking changes.